test/src/tests-identityApiClient.ts

```diff
@@ -0,0 +1,480 @@
+            expect(parseIdentityResponseSpy.args[0][3]).to.deep.equal(originalIdentityApiData);
+            expect(parseIdentityResponseSpy.args[0][4]).to.equal('identify');
+            expect(parseIdentityResponseSpy.args[0][5]).to.deep.equal(identityRequest.known_identities);
+            expect(parseIdentityResponseSpy.args[0][6]).to.equal(false);
+
+            window.fetch = fetch;
+        });
+
+        it('should construct the correct fetch payload', async () => {
+            fetchMockSuccess(urls.identify, apiSuccessResponseBody);
+
+            const callbackSpy = sinon.spy();
+
+            const mpInstance: MParticleWebSDK = ({
+                Logger: {
+                    verbose: () => {},
+                    error: () => {},
+                },
+                _Helpers: {
+                    createServiceUrl: () =>
+                        'https://identity.mparticle.com/v1/',
+
+                    invokeCallback: () => {},
+                },
+                _Store: {
+                    devToken: 'test_key',
+                    SDKConfig: {
+                        identityUrl: '',
+                    },
+                },
+                _Persistence: {},
+            } as unknown) as MParticleWebSDK;
+
+            const identityApiClient: IIdentityApiClient = new IdentityAPIClient(
+                mpInstance
+            );
+
+            const parseIdentityResponseSpy = sinon.spy();
+
+            await identityApiClient.sendIdentityRequest(
+                identityRequest,
+                'identify',
+                callbackSpy,
+                originalIdentityApiData,
+                parseIdentityResponseSpy,
+                testMPID,
+                identityRequest.known_identities
+            );
+
+            const expectedFetchPayload = {
+                method: 'post',
+                headers: {
+                    Accept: 'text/plain;charset=UTF-8',
+                    'Content-Type': 'application/json',
+                    'x-mp-key': 'test_key',
+                },
+                body: JSON.stringify(identityRequest),
+            };
+
+            expect(fetchMock.calls()[0][1].method).to.deep.equal(expectedFetchPayload.method);
+            expect(fetchMock.calls()[0][1].body).to.deep.equal(expectedFetchPayload.body);
+            expect(fetchMock.calls()[0][1].headers).to.deep.equal(expectedFetchPayload.headers);
+        });
+    });
+
+    describe('#sendAliasRequest', () => {
+        const aliasUrl = 'https://jssdks.mparticle.com/v1/identity/test_key/Alias';
+
+        beforeEach(function() {
+            fetchMock.post(urls.events, 200);
+            mockServer = sinon.createFakeServer();
+            mockServer.respondImmediately = true;
+
+            mockServer.respondWith(urls.identify, [
+                200,
+                {},
+                JSON.stringify({ mpid: testMPID, is_logged_in: false }),
+            ]);
```

Comment:
mock server is for xhr.  and seems like xhr is not being tested here, only fetchMock.  we should either add the xhr legacy tests, or remove.  i'd vote for the former just because there aren't many tests here.

Reply:
This collection of tests was copied directly from the existing alias request tests. I guess we missed it when we refactored.